detectors/aws/eks/eks_resource_detector.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	clusterName, err := getClusterName(detector.utils)
+	if hasProblem(err) {
+		return nil, err
+	}
+	if clusterName != "" {
+		labels = append(labels, semconv.K8SClusterNameKey.String(clusterName))
+	}
+
+	containerID, err := detector.utils.getContainerID()
+	if hasProblem(err) {
+		return nil, err
+	}
+	if containerID != "" {
+		labels = append(labels, semconv.ContainerIDKey.String(containerID))
+	}
+
+	return resource.New(labels...), nil
+
+}
+
+func isEks(utils detectorUtils) (bool, error) {
+	if !isK8s(utils) {
+		return false, nil
+	}
+
+	awsAuth, err := utils.fetchString("GET", k8sSvcURL+authConfigmapPath)
+	if hasProblem(err) {
+		return false, err
+	}
+
+	return awsAuth != "", nil
+}
+
+func isK8s(utils detectorUtils) bool {
+	return utils.fileExists(k8sTokenPath) && utils.fileExists(k8sCertPath)
+}
+
+// eksDetectorUtils is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
+	fmt.Println("REAL HIT")
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// fetchString is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
+	request, err := http.NewRequest(httpMethod, URL, nil)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	authHeader, err := getK8sCredHeader()
+	if hasProblem(err) {
+		return "", err
+	}
+	request.Header.Set("Authorization", authHeader)
+
+	caCert, err := ioutil.ReadFile(k8sCertPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	client := &http.Client{
+		Timeout: millisecondTimeOut * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	response, err := client.Do(request)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func getK8sCredHeader() (string, error) {
+	content, err := ioutil.ReadFile(k8sTokenPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	return "Bearer " + string(content), nil
+}
+
+func getClusterName(utils detectorUtils) (string, error) {
+	resp, err := utils.fetchString("GET", k8sSvcURL+cwConfigmapPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	var parsedResp JSONResponse
+	err = json.Unmarshal([]byte(resp), &parsedResp)
+	if hasProblem(err) {
+		return "", err
+	}
+	clusterName := parsedResp.Data.ClusterName
+
+	return clusterName, nil
+}
+
+// getContainerID is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) getContainerID() (string, error) {
+	content, err := readFile(defaultCgroupPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	for _, line := range content {
+		if lineLength := len(line); lineLength > containerIDLength {
+			return line[lineLength-containerIDLength:], nil
```

Comment:
always put spaces between variables and `-`, `=`, `+` (most operators).

```
return line[lineLength - containerIDLength:], nil
```

Makes it more readable

Reply:
I realized VS code deletes whitespace around operators when I save files. Have you encountered this issue before? For now, I've switched to using a different IDE since I can't find the setting to turn it off.

Reply:
that's weird... properly configured, VS code should help you format your code according to Go style...

Unless of course I am wrong, and putting spaces between operators in this case is not the standard practice.

Reply:
It looks like I might am probably wrong: https://github.com/golang/go/issues/1206